tests/func/test_get.py

```diff
@@ -75,6 +107,55 @@ def test_get_a_dvc_file(erepo):
         Repo.get(erepo.root_dir, "some_file.dvc")
 
 
+# https://github.com/iterative/dvc/pull/2837#discussion_r352123053
+def test_get_full_dvc_path(erepo):
+    external_data_dir = erepo.mkdtemp()
+    external_data = os.path.join(external_data_dir, "ext_data")
+    with open(external_data, "w+") as fobj:
+        fobj.write("ext_data")
+
+    cur_dir = os.getcwd()
+    os.chdir(erepo.root_dir)
+    erepo.dvc.add(external_data)
+    erepo.dvc.scm.add(["ext_data.dvc"])
+    erepo.dvc.scm.commit("add external data")
+    os.chdir(cur_dir)
+
+    Repo.get(erepo.root_dir, external_data, "ext_data_imported")
+    assert os.path.isfile("ext_data_imported")
+    assert filecmp.cmp(external_data, "ext_data_imported", shallow=False)
+
+
+def test_non_cached_output(tmp_path, erepo):
+    os.chdir(erepo.root_dir)
+    erepo.dvc.run(
+        outs_no_cache=["non_cached_file"], cmd="echo hello > non_cached_file"
+    )
+    erepo.dvc.scm.add(["non_cached_file", "non_cached_file.dvc"])
+    erepo.dvc.scm.commit("add non-cached output")
+    os.chdir(tmp_path)
```

Comment:
Looks like you simply need to
```suggestion
    os.chdir(fspath_py35(tmp_path))
```
since os.chdir started accepting PathLike starting from 3.6